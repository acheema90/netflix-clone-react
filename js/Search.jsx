import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setSearchTerm } from './actionCreators';
import ShowCard from './ShowCard';
import Header from './Header';

class Search extends Component {
    constructor(props) {
        super(props);
        this.handleSearchTermChange = this.props.handleSearchTermChange;
    }

    // every react Component class must have a render method
    // and the render method must return markup
    render() {
        return (
            <div className='search'>
                <Header
                    showSearch={true}
                    searchTerm={this.props.searchTerm}
                    onChangeHandler={this.handleSearchTermChange}
                />
                <div>
                    {this.props.shows
                        .filter(show =>
                            `${show.title} ${show.description}`
                            .toLowerCase()
                            .indexOf(this.props.searchTerm.toLowerCase()) !== -1
                        )
                        .map(show => <ShowCard key={show.imdbID} show={show} /> )}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return { searchTerm: state.searchTerm };
};

const mapDispatchToProps = (dispatch) => {
    return({
        handleSearchTermChange(event) {
            dispatch(setSearchTerm(event.target.value));
        }
    });
};

export default connect(mapStateToProps, mapDispatchToProps)(Search);
