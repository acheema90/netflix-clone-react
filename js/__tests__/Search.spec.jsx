import React from 'react';
import { shallow } from 'enzyme';
import preload from '../../data';
import Search, { Unwrapped as UnwrappedSearch} from '../Search';
import ShowCard from '../ShowCard';

// jest gives us istanbul coverage for free!

test('Search renders correctly', () => {
    const component = shallow(<UnwrappedSearch  shows={preload.shows}
         searchTerm=''/>);
    expect(component).toMatchSnapshot();
});

test('Search should render correct amount of shows', () => {
    const component = shallow(<UnwrappedSearch shows={preload.shows}
         searchTerm=''/>);
    expect(component.find(ShowCard).length)
        .toEqual(preload.shows.length);
});

test('Search should render correct amount of shows based on search term', () => {
    const searchTerm = 'black';
    const component = shallow(<UnwrappedSearch shows={preload.shows}
         searchTerm={searchTerm}/>);
    // component.find('input').simulate('change', { target: { value:searchTerm }});
    const showCount = preload.shows.filter(show =>
        `${show.title} ${show.description}`
        .toLowerCase()
        .indexOf(searchTerm.toLowerCase()) !== -1
    );
    expect(component.find(ShowCard).length)
        .toEqual(showCount.length);
});
