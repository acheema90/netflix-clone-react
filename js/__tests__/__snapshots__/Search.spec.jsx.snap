// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <Header
    searchTerm=""
    showSearch={true}
  />
  <div>
    <ShowCard
      show={
        Object {
          "description": "Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; \\"Earnest 'Earn' Marks,\\" an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.",
          "imdbID": "tt4288182",
          "poster": "a.jpg",
          "title": "Atlanta",
          "trailer": "https://www.youtube-nocookie.com/embed/MpEdJ-mmTlY?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2008–2013",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "U.S. Attorney Chuck Rhoades goes after hedge fund king, Bobby \\"Axe\\" Axelrod in a battle between two powerful New York figures.",
          "imdbID": "tt4270492",
          "poster": "b.jpg",
          "title": "Billions",
          "trailer": "https://www.youtube-nocookie.com/embed/_raEUMLL-ZI?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2016–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A television anthology series that shows the dark side of life and technology.",
          "imdbID": "tt2085059",
          "poster": "bm.jpg",
          "title": "Black Mirror",
          "trailer": "https://www.youtube-nocookie.com/embed/jDiYGjp5iFg?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2011–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A high school chemistry teacher diagnosed with inoperable lung cancer turns to manufacturing and selling methamphetamine in order to secure his family's future.",
          "imdbID": "tt0903747",
          "poster": "bb.jpg",
          "title": "Breaking Bad",
          "trailer": "https://www.youtube-nocookie.com/embed/XZ8daibM3AE?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2008–2013",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years.",
          "imdbID": "tt0944947",
          "poster": "got.jpg",
          "title": "Game of Thrones",
          "trailer": "https://www.youtube-nocookie.com/embed/giYeaKsXnsI?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2011–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A bipolar CIA operative becomes convinced a prisoner of war has been turned by al-Qaeda and is planning to carry out a terrorist attack on American soil.",
          "imdbID": "tt1796960",
          "poster": "h.jpg",
          "title": "Homeland",
          "trailer": "https://www.youtube-nocookie.com/embed/KyFmS3wRPCQ?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2011–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A Congressman works with his equally conniving wife to exact revenge on the people who betrayed him.",
          "imdbID": "tt1856010",
          "poster": "hoc.jpg",
          "title": "House of Cards",
          "trailer": "https://www.youtube-nocookie.com/embed/NTzycsqxYJ0?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2013–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "The personal and professional life of Dev, a 30-year-old actor in New York.",
          "imdbID": "tt4635276",
          "poster": "mon.jpg",
          "title": "Master of None",
          "trailer": "https://www.youtube-nocookie.com/embed/6bFvb3WKISk?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2015–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A chronicled look at the criminal exploits of Colombian drug lord Pablo Escobar.",
          "imdbID": "tt2707408",
          "poster": "n.jpg",
          "title": "Narcos",
          "trailer": "https://www.youtube-nocookie.com/embed/U7elNhHwgBU?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2015–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "The story of Piper Chapman, a woman in her thirties who is sentenced to fifteen months in prison after being convicted of a decade-old crime of transporting money for her drug-dealing girlfriend.",
          "imdbID": "tt2372162",
          "poster": "oitnb.jpg",
          "title": "Orange Is the New Black",
          "trailer": "https://www.youtube-nocookie.com/embed/th8WT_pxGqg?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2013–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper.",
          "imdbID": "tt2575988",
          "poster": "sv.jpg",
          "title": "Silicon Valley",
          "trailer": "https://www.youtube-nocookie.com/embed/69V__a49xtw?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2014–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "When a young boy disappears, his mother, a police chief, and his friends must confront terrifying forces in order to get him back.",
          "imdbID": "tt4574334",
          "poster": "st.jpg",
          "title": "Stranger Things",
          "trailer": "https://www.youtube-nocookie.com/embed/9Egf5U8xLo8?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2016–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Two Soviet intelligence agents pose as a married couple to spy on the American government.",
          "imdbID": "tt2149175",
          "poster": "ta.jpg",
          "title": "The Americans",
          "trailer": "https://www.youtube-nocookie.com/embed/HjuUkbhsI24?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2013–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Former Senator Selina Meyer finds that being Vice President of the United States is nothing like she hoped and everything that everyone ever warned her about.",
          "imdbID": "tt1759761",
          "poster": "v.jpg",
          "title": "Veep",
          "trailer": "https://www.youtube-nocookie.com/embed/I2HoCG_iyhU?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2012–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence.",
          "imdbID": "tt0475784",
          "poster": "ww.jpg",
          "title": "Westworld",
          "trailer": "https://www.youtube-nocookie.com/embed/eX3u0IlBBO4?rel=0&amp;control=0&amp;showinfo=0",
          "year": "2016–",
        }
      }
    />
  </div>
</div>
`;
